### ISSUE TITLE
About tilemaps with out-of-bounds tile references

### ISSUE THREAD

**Issue by dacap** [aseprite/aseprite#4119]
How should we handle tilemaps that have tile references that are pointing to indexes larger than the current tileset size?

In #3958 we can see how reducing the tileset size will create a tilemap referencing tile numbers/indexes that are not part of the tileset anymore. This brings several issues that are not easy to fix together:

1. The bug in the same (#3958), where after painting in Auto mode the modified tiles are updated correctly, but the untouched areas of the tilemap still reference the old indexes (now new/valid modified tiles) instead of referencing empty tiles (and this happens for all existing tilemaps with out-of-bounds tile references)
2. After #4093 to fix a crash moving these out-of-bounds tile references, in someway a regression was introduced (#4105) but anyway the Auto mode never worked correctly as we pointed above
3. After ed8e2c8392b01fbe844bdab11a912a6f298c2d1d fix, loading a `.aseprite` with out-of-bounds tile references, will make that those references are just ignored, so the information is lost anyway (the out-of-bounds tile references can be kept only on memory or in the file in the first save, but after loading and saving it again, they will be lost). This is [already fixed](https://github.com/aseprite/aseprite/pull/3967#issuecomment-1640086600) in my `flip` branch but probably I’ll cherry-pick in `main`.

We can think that we should update/increase all out-of-bounds tile references the number of tiles that were added/deleted in the tileset, to keep the information alive, but do we really need to keep this information alive? Sometimes if we switch between tilesets it would probably be required to keep the information temporarily (so tilemaps are not modified when we switch to another tileset), but I’m still not sure how to handle all the cases properly and make the situation transparent to the user.

Probably we can identify two kind of users:

1. The one that don’t want to deal with this, and the out-of-bounds references are just a mistake/no desirable (generally using Auto mode)
2. The one that want to keep all tile references as they are, even if they reference something out-of-bounds (generally using Manual mode)

Mainly for the 2nd kind of user, we need to offer some better tooling, e.g. show tilemap edges, a way to change the tilemap bounds, don’t shrink it automatically, etc.

We’ll kick all this forward for a possible v1.3.1 or future v1.3.x to avoid delaying v1.3 more.
